gdal/swig/python/setup.py

```diff
@@ -142,34 +115,16 @@ def fetch_config(option, gdal_config='gdal-config'):
 
     command = gdal_config + " --%s" % option
 
+    import subprocess
+    command, args = command.split()[0], command.split()[1]
     try:
-        import subprocess
-        command, args = command.split()[0], command.split()[1]
-        from sys import version_info
-        if version_info >= (3, 0, 0):
-            try:
-                p = subprocess.Popen([command, args], stdout=subprocess.PIPE)
-            except OSError:
-                e = sys.exc_info()[1]
-                raise gdal_config_error(e)
-            r = p.stdout.readline().decode('ascii').strip()
-        else:
-            exec("""try:
-    p = subprocess.Popen([command, args], stdout=subprocess.PIPE)
-except OSError, e:
-    raise gdal_config_error, e""")
-            r = p.stdout.readline().strip()
-        p.stdout.close()
-        p.wait()
-
-    except ImportError:
-
-        import popen2
-
-        p = popen2.popen3(command)
-        r = p[0].readline().strip()
-        if not r:
-            raise Warning(p[2].readline())
+        p = subprocess.Popen([command, args], stdout=subprocess.PIPE)
+    except OSError:
+        e = sys.exc_info()[1]
+        raise gdal_config_error(e)
+    r = p.stdout.readline().decode('ascii').strip()
+    p.stdout.close()
+    p.wait()
 
     return r
```

Comment:
```suggestion
        p = subprocess.run([command, args], stdout=subprocess.PIPE, check=True, encoding='ascii')
        return p.stdout.strip()
    except OSError as e:
        raise gdal_config_error(str(e)) from e
```

Reply:
There are a whole bunch of these; I'd rather clean them all up in one go.